contracts/contracts/vault/VaultCore.sol

```diff
@@ -74,25 +76,22 @@ contract VaultCore is VaultStorage {
             price = 1e8;
         }
         require(price >= MINT_MINIMUM_ORACLE, "Asset price below peg");
-        uint256 assetDecimals = Helpers.getDecimals(_asset);
         // Scale up to 18 decimal
-        uint256 unitAdjustedDeposit = _amount.scaleBy(18, assetDecimals);
-        uint256 priceAdjustedDeposit = _amount.mulTruncateScale(
-            price.scaleBy(18, 8), // Oracles have 8 decimal precision
-            10**assetDecimals
-        );
+        uint256 priceAdjustedDeposit = (_toUnits(_amount, _asset) * price) /
```

Comment:
much cleaner 👍

Reply:
Hmm also for `rEth` that is a token with an `getExchangeRate` wont this erroneously adjust the price twice? Since the `_toUnits` function already multiplies the raw amount with the exchange rate of the token lower in this file -> `return (raw * IExchangeRateToken(_asset).exchangeRate()) / 1e18; `